### PR TITLE
Bump images to track with upstream

### PR DESCRIPTION
**Pull request by brandond**
Also add comments to note where upstream versions can be found

#### Pull Request Checklist ####

- [x] Any new images or tags consumed by charts has been added [here](https://github.com/rancher/image-mirror)
- [x] Chart version has been incremented (if necessary)
- [x] That helm lint and pack run successfully on the chart.
- [ ] Deployment of the chart has been tested and verified that it functions as expected.
- [x] Changes to scripting or CI config have been tested to the best of your ability

#### Types of Change ####

version bump

#### Linked Issues ####

* 

#### After the PR is merged ####

Once the PR is merged, typically upon a new release, the necessary teams will be notified via Slack hook to perform the RKE2 Charts and RKE2 changes. Any developer working on this issue is not responsible for updating RKE2 Charts or RKE2.

### CODE REVIEW

File: charts/rancher-vsphere-cpi/values.yaml [hunk 1]
```diff
@@ -81,7 +85,7 @@ versionOverrides:
 
 cloudControllerManager:
   repository: rancher/mirrored-cloud-provider-vsphere-cpi-release-manager
-  tag: v1.22.7
+  tag: latest
```
Comment (brandond) [inline charts/rancher-vsphere-cpi/values.yaml:88]: Set the default tag to `latest` to make it obvious when the version overrides are not working properly

File: charts/rancher-vsphere-csi/values.yaml [hunk 2]
```diff
@@ -25,22 +25,22 @@ csiController:
     enabled: false
   image:
     repository: rancher/mirrored-cloud-provider-vsphere-csi-release-driver
-    tag: v2.6.2
+    tag: latest
```
Comment (brandond) [inline charts/rancher-vsphere-csi/values.yaml:28]: Set the default tags to `latest` to make it obvious when the version overrides are not working properly

File: charts/rancher-vsphere-cpi/templates/role-binding.yaml [hunk 3]
```diff
@@ -1,43 +1,40 @@
 {{- if .Values.cloudControllerManager.rbac.enabled -}}
-apiVersion: v1
-kind: List
-metadata: {}
```
Comment (brandond) [inline charts/rancher-vsphere-cpi/templates/role-binding.yaml:4]: list with empty metadata, while valid, was breaking `helm lint`

**Review by jiaqiluo — APPROVED** [review]

**Comment by a-blender** [comment]
There are new CPI versions v1.25.2 and v1.24.5 but they are 2-3 weeks new and are not in `image-mirror` yet so can be bumped to the next release. Relevant issue [here](https://github.com/rancher/rancher/issues/41626)!

**Pull request merged**